pkg/scalers/selenium_grid_scaler.go

```diff
@@ -0,0 +1,201 @@
+	v, err := s.getSessionsCount()
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, fmt.Errorf("error requesting selenium grid endpoint: %s", err)
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *v,
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *seleniumGridScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	targetValue := resource.NewQuantity(s.metadata.targetValue, resource.DecimalSI)
+	metricName := kedautil.NormalizeString(fmt.Sprintf("%s-%s", "seleniumgrid", s.metadata.browserName))
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: metricName,
+		},
+		Target: v2beta2.MetricTarget{
+			Type:         v2beta2.AverageValueMetricType,
+			AverageValue: targetValue,
+		},
+	}
+	metricSpec := v2beta2.MetricSpec{
+		External: externalMetric, Type: externalMetricType,
+	}
+	return []v2beta2.MetricSpec{metricSpec}
+}
+
+func (s *seleniumGridScaler) IsActive(ctx context.Context) (bool, error) {
+	v, err := s.getSessionsCount()
+	if err != nil {
+		return false, err
+	}
+
+	return v.AsApproximateFloat64() > 0.0, nil
+}
+
+func (s *seleniumGridScaler) getSessionsCount() (*resource.Quantity, error) {
+	body, err := json.Marshal(map[string]string{
+		"query": "{ sessionsInfo { sessionQueueRequests, sessions { id, capabilities, nodeId } } }",
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("POST", s.metadata.url, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := s.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode != http.StatusOK {
+		msg := fmt.Sprintf("selenium grid returned %d", res.StatusCode)
+		return nil, errors.New(msg)
+	}
+
+	defer res.Body.Close()
+	b, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	v, err := getCountFromSeleniumResponse(b, s.metadata.browserName)
+	if err != nil {
+		return nil, err
+	}
+	return v, nil
+}
+
+func getCountFromSeleniumResponse(b []byte, browserName string) (*resource.Quantity, error) {
+	var count int64 = 0
+	var seleniumResponse seleniumResponse = seleniumResponse{}
+
+	if err := json.Unmarshal(b, &seleniumResponse); err != nil {
+		return nil, err
+	}
+
+	var sessionQueueRequests = seleniumResponse.Data.SessionsInfo.SessionQueueRequests
+	for _, sessionQueueRequest := range sessionQueueRequests {
+		var capability = capability{}
+		if err := json.Unmarshal([]byte(sessionQueueRequest), &capability); err == nil {
+			if capability.BrowserName == browserName {
+				count = count + 1
+			}
+		}
+	}
+
+	var sessions = seleniumResponse.Data.SessionsInfo.Sessions
+	for _, session := range sessions {
+		var capability = capability{}
+		if err := json.Unmarshal([]byte(session.Capabilities), &capability); err == nil {
```

Comment:
what happens if `err != nil` ?

Reply:
We do not want the count to be influenced when error is not nil. Your thoughts.

Reply:
I see, would be nice to at least log this.